### PR TITLE
add panel for execution overhead alert

### PR DESCRIPTION
**Pull request by ramessesii2**
- add query for PipelineRun execution alert to the alert panel

Preview: 
image

### CODE REVIEW

**Review by gabemontero — CHANGES_REQUESTED** [review]
a couple of minor updates @ramessesii2

File: operator/gitops/argocd/grafana/dashboards/pipeline-service-dashboard.json [hunk 1]
```diff
@@ -165,6 +165,75 @@
...
+            "uid": "${DS_PROMETHEUS-APPSTUDIO-DS}"
+          },
+          "editorMode": "code",
+          "expr": "(\n  (sum(pipelinerun_gap_between_taskruns_milliseconds_sum)/1000 - sum(pipelinerun_gap_between_taskruns_milliseconds_sum offset 30m)/1000)\n  / \n  (sum(pipelinerun_gap_between_taskruns_milliseconds_count) - sum(pipelinerun_gap_between_taskruns_milliseconds_count offset 30m))\n) \n/ \n(\n  (sum(tekton_pipelines_controller_pipelinerun_duration_seconds_sum{status='success'}) - sum(tekton_pipelines_controller_pipelinerun_duration_seconds_sum{status='success'} offset 30m))\n  / \n  (sum(tekton_pipelines_controller_pipelinerun_duration_seconds_count{status='success'}) - sum(tekton_pipelines_controller_pipelinerun_duration_seconds_count{status='success'}offset 30m))\n)",
+          "legendFormat": "__auto",
+          "range": true,
+          "refId": "A"
+        }
+      ],
+      "title": "Execution",
```
Comment (gabemontero) [inline operator/gitops/argocd/grafana/dashboards/pipeline-service-dashboard.json:234]:
```suggestion
      "title": "Tekton Execution Overhead",
```

And can you change the scheduling one to `Tekton Scheduling Overhead`

thanks

File: operator/gitops/argocd/grafana/dashboards/pipeline-service-dashboard.json [hunk 2]
```diff
@@ -165,6 +165,75 @@
...
+      },
+      "pluginVersion": "9.1.6",
+      "targets": [
+        {
+          "datasource": {
+            "type": "prometheus",
+            "uid": "${DS_PROMETHEUS-APPSTUDIO-DS}"
+          },
+          "editorMode": "code",
+          "expr": "(\n  (sum(pipelinerun_gap_between_taskruns_milliseconds_sum)/1000 - sum(pipelinerun_gap_between_taskruns_milliseconds_sum offset 30m)/1000)\n  / \n  (sum(pipelinerun_gap_between_taskruns_milliseconds_count) - sum(pipelinerun_gap_between_taskruns_milliseconds_count offset 30m))\n) \n/ \n(\n  (sum(tekton_pipelines_controller_pipelinerun_duration_seconds_sum{status='success'}) - sum(tekton_pipelines_controller_pipelinerun_duration_seconds_sum{status='success'} offset 30m))\n  / \n  (sum(tekton_pipelines_controller_pipelinerun_duration_seconds_count{status='success'}) - sum(tekton_pipelines_controller_pipelinerun_duration_seconds_count{status='success'}offset 30m))\n)",
```
[the remark this answers is not in the document]
Comment (gabemontero) [reply operator/gitops/argocd/grafana/dashboards/pipeline-service-dashboard.json:228]: per my find in https://redhat-internal.slack.com/archives/C032EJ007C0/p1696862756522219 looks like this is not uncommon with grafana and there is a way to force it to say "No data" if there is in fact no data.

[the remark this answers is not in the document]
Comment (gabemontero) [reply operator/gitops/argocd/grafana/dashboards/pipeline-service-dashboard.json:228]: add the same thing ^^ to this panel @ramessesii2

[the remark this answers is not in the document]
Comment (ramessesii2) [reply operator/gitops/argocd/grafana/dashboards/pipeline-service-dashboard.json]: done

File: operator/gitops/argocd/grafana/dashboards/pipeline-service-dashboard.json (hunk 1 above)
Comment (ramessesii2) [reply operator/gitops/argocd/grafana/dashboards/pipeline-service-dashboard.json]: I've changed the title as well.

File: operator/gitops/argocd/grafana/dashboards/pipeline-service-dashboard.json [hunk 3]
```diff
@@ -165,6 +165,75 @@
...
+          "thresholds": {
+            "mode": "absolute",
+            "steps": [
+              {
+                "color": "green",
+                "value": null
+              },
+              {
+                "color": "red",
+                "value": 80
```
[the remark this answers is not in the document]
Comment (ramessesii2) [reply operator/gitops/argocd/grafana/dashboards/pipeline-service-dashboard.json]: done. Completely missed it.

**Comment by ramessesii2** [comment]
The CI failure with static-code-analysis:
```
- ./operator/gitops/argocd/grafana/dashboards/pipeline-service-dashboard.json: FAIL
[ERROR] 'datasource' element(s) found in './operator/gitops/argocd/grafana/dashboards/pipeline-service-dashboard.json'
```
should be fixed with #807 I believe.

File: operator/gitops/argocd/grafana/dashboards/pipeline-service-dashboard.json [hunk 4]
```diff
@@ -289,8 +359,7 @@
                 "mode": "absolute",
                 "steps": [
                   {
-                    "color": "green",
-                    "value": null
```
Comment (gabemontero) [inline operator/gitops/argocd/grafana/dashboards/pipeline-service-dashboard.json:293]: hmm ... assumed the export did this .... I suppose it is OK

let's try to verify that @ramessesii2 after this merges

**Review by gabemontero — APPROVED** [review]
minus my one comment about the export removing `value: null` which we can sort out after merge and in pipeline-service / staging I'm good with merging

**Comment by gabemontero** [comment]
> did you rebase your PR @ramessesii2 ? I'm not seeing a force push since your last comment that would indicate so
> 
> the dashboard validation error still occurred

@Roming22 got the rebase button added ... trying it now

File: operator/gitops/argocd/grafana/dashboards/pipeline-service-dashboard.json (hunk 4 above)
Comment (ramessesii2) [reply operator/gitops/argocd/grafana/dashboards/pipeline-service-dashboard.json:293]: yes, that's what I got when exported.

File: operator/gitops/argocd/grafana/dashboards/pipeline-service-dashboard.json [hunk 5]
```diff
@@ -166,6 +166,76 @@
       "transformations": [],
       "type": "stat"
     },
+    {
+      "datasource": {
+        "type": "prometheus",
+        "uid": "${DS_PROMETHEUS-APPSTUDIO-DS}"
+      },
+      "description": "The proportion of time spent as gaps between TaskRuns relative to the overall duration of execution of successful PipelineRuns",
```
Comment (enarha) [inline operator/gitops/argocd/grafana/dashboards/pipeline-service-dashboard.json:174]:
```suggestion
      "description": "Time gaps between TaskRuns relative to the overall duration of successful PipelineRuns",
```

**Review by Roming22 — APPROVED** [review]

**Pull request merged**